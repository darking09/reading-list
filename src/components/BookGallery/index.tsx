import Book from '@components/Book'
import BookGalleryWrapper from '@components/BookGalleryWrapper'
import { Book as BookType } from '@typesFiles/Books'
import { BookGalleryProps } from '@typesFiles/props/bookGallery'

export default function BookGallery({ books } : BookGalleryProps ) {

  return (
    <section className="row-start-1 col-span-3">
      <div className="grid grid-flow-row-dense grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 2xl:grid-cols-6 p-2">
        {
          books.map((b : BookType) => (
            <BookGalleryWrapper key={b.book.ISBN} book={b}>
              <Book bookContainer={b} />
            </BookGalleryWrapper>
          ))
        }
      </div>
    </section>
  )
}
