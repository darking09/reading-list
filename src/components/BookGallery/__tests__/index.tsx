/* eslint-disable react/display-name */
import BookGallery from "../index"
import { render } from "@testing-library/react"

jest.mock('@components/BookGalleryWrapper', () => ({ children } : any) => <div>{children}</div>)
jest.mock('@components/Book', () => ({ children } : any) => <div>{children}</div>)

describe('BookGallery', () => {
  it('should render', () => {
    const { container } = render(
      <BookGallery
        books={[
          {
            id: 1,
            pages: 1,
            genre_id: 1,
            author_id: {
              name: 'name',
              otherBooks: [],
            },
            synopsis: 'synopsis',
            year: 1,
            ISBN: 'ISBN',
            is_available: true,
            title: 'title',
            cover: 'cover',
            created_at: new Date(),
          },
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
