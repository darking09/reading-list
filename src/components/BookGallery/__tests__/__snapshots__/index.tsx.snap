// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookGallery should render 1`] = `
<div>
  <section
    class="row-start-1 col-span-3 h-auto overflow-auto"
  >
    <div
      class="grid grid-flow-row-dense grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 2xl:grid-cols-6 p-2"
    >
      <div>
        <div />
      </div>
    </div>
  </section>
</div>
`;
